### ISSUE TITLE
Using Result type from FSharp.Core 4.4.1.0 with F# 4.0 compiler causes runtime error

### ISSUE THREAD

**Issue by vasily-kirichenko** [Microsoft/visualfsharp#3354]
1. Create a console project in VS 2015
2. Add FSharp.Core NuGet package reference via Paket
3. Use `Result` type like this:
```fsharp
[<EntryPoint>]
let main _ = 
    let result = Result.Ok "Hello"
    printfn "%A" result
    0
```
4. Run the console:

![image](https://user-images.githubusercontent.com/873919/28408084-f7b0f57c-6d3e-11e7-8037-254b2c4b7536.png)

Known workaround: reference FSharp.Compiler.Tools package as well, it brings F# 4.1 compiler.

**Comment by dsyme** [comment]
Yes, this is an unfortunate problem with F# 4.0 using FSHarp.Core 4.4.1.0 that we did not predict or consider in [the RFC](https://github.com/fsharp/fslang-design/blob/master/FSharp-4.1/FS-1004-result-type.md).  There is little we can do about it now since we are not going to update Visual Studio 2015, and as you mention there is a workaround.

I have [updated the RFC template for future RFCs](https://github.com/fsharp/fslang-design/pull/215) to make sure we consider consumption of new FSHarp.Core from down-level versions of F# whenever we are introducing new constructs to FSHarp.Core.

It is interesting to ask ourselves 
* would we have delayed the addition of a value type Result to FSharp.Core by one or two versions because of this problem?
* are there things we can be adding to the compiler now in order to make way for new features in the future?
